Enforce LF line endings

zouharvi opened the pull request:
As per discussion in #111 this PR:
- adds the attribute to `.gitignore` that informs git to automatically transform detected text files to LF;
- applies that rule to all existing text files in the repository.

(Should be merged after #171.)

zouharvi commented:
I'll need to rebase it after the other PR since they'd be conflicting.

snukky commented:
I will wait with merging this change until we have other pending PRs and WIP merged as it may lead to merging issues.

zouharvi commented:
Now that there is less activity on this repo and this change is thus less breaking, I redid the LF conversion and added the attribute to made new files be LF automatically.

In addition the the previous comment, I also added `--strip-trailing-cr` to the `diff` command because `csv.writer` outputs in CRLF, likely for backward compatibility reasons.

@snukky The tests seem to be failing because of some tooling issue. Locally it looks ok:
```
$ bash RegressionTests/run.sh
[11/25/2024 13:25:02] Appraise Git commit: 103e86d
[11/25/2024 13:25:02] Appraise root directory: /home/vilda/Appraise
[11/25/2024 13:25:02] Python executable: /home/vilda/miniconda3/bin/python3
[11/25/2024 13:25:02] Python version: Python 3.11.5
[11/25/2024 13:25:02] Using database: development.db
/home/vilda/Appraise/Appraise/settings.py:34: UserWarning: Using the default SECRET_KEY value! Set and export APPRAISE_SECRET_KEY envvar instead
  warnings.warn(
Operations to perform:
  Apply all migrations: Campaign, Dashboard, EvalData, admin, auth, contenttypes, sessions
Running migrations:
  No migrations to apply.
/home/vilda/Appraise/Appraise/settings.py:34: UserWarning: Using the default SECRET_KEY value! Set and export APPRAISE_SECRET_KEY envvar instead
  warnings.warn(
/home/vilda/Appraise/Appraise/settings.py:34: UserWarning: Using the default SECRET_KEY value! Set and export APPRAISE_SECRET_KEY envvar instead
  warnings.warn(
Superuser created successfully.
[11/25/2024 13:25:05] Running regression tests...
[11/25/2024 13:25:05] Checking directory: /home/vilda/Appraise/RegressionTests/tests
[11/25/2024 13:25:05] Checking directory: /home/vilda/Appraise/RegressionTests/tests/examples
[11/25/2024 13:25:05] Running RegressionTests/tests/examples/test_examples_data.sh ... OK
[11/25/2024 13:25:23] Test took 00:00:17.912s
[11/25/2024 13:25:23] Running RegressionTests/tests/examples/test_examples_direct.sh ... OK
[11/25/2024 13:25:32] Test took 00:00:9.536s
[11/25/2024 13:25:32] Running RegressionTests/tests/examples/test_examples_mqmesa.sh ... OK
[11/25/2024 13:25:39] Test took 00:00:7.041s
[11/25/2024 13:25:39] Running RegressionTests/tests/examples/test_examples_named_entities.sh ... OK
[11/25/2024 13:25:49] Test took 00:00:9.335s
[11/25/2024 13:25:49] Running RegressionTests/tests/examples/test_examples_pairwise.sh ... OK
[11/25/2024 13:25:57] Test took 00:00:8.602s
[11/25/2024 13:25:57] Running RegressionTests/tests/examples/test_examples_pairwise_critical_errors.sh ... OK
[11/25/2024 13:26:06] Test took 00:00:8.657s
[11/25/2024 13:26:06] Running RegressionTests/tests/examples/test_examples_pairwise_document.sh ... OK
[11/25/2024 13:26:15] Test took 00:00:9.314s
[11/25/2024 13:26:15] Checking directory: /home/vilda/Appraise/RegressionTests/tests/examples/Batches
[11/25/2024 13:26:15] Checking directory: /home/vilda/Appraise/RegressionTests/tests/createtasks
[11/25/2024 13:26:15] Running RegressionTests/tests/createtasks/test_create_wmt22_selected_docs.sh ... OK
[11/25/2024 13:26:15] Test took 00:00:0.124s
[11/25/2024 13:26:15] Running RegressionTests/tests/createtasks/test_create_wmt22_tasks.sh ... OK
[11/25/2024 13:26:16] Test took 00:00:0.222s
---------------------
Ran 9 tests in 00:01:10.860s, 9 passed, 0 failed
```

zouharvi commented:
@snukky Also fixed the testing after v2 deprecation. Was a really easy change.

https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/


I'll merge this because there are not other active PRs (apart from #161?) and my other PRs will depend on this. If there is ever a conflict from some older PR, just make sure that everything is LF: `find . -type f -exec dos2unix {} \;`
